src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionServiceImpl.java

```diff
@@ -1,24 +1,81 @@
 import org.folio.innreach.repository.InnReachTransactionRepository;
 import org.springframework.stereotype.Service;
 
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.Set;
+import java.util.UUID;
+
+import static org.folio.innreach.domain.dto.folio.inventory.InventoryItemStatus.AVAILABLE;
+import static org.folio.innreach.domain.dto.folio.inventory.InventoryItemStatus.AWAITING_DELIVERY;
+import static org.folio.innreach.domain.dto.folio.inventory.InventoryItemStatus.AWAITING_PICKUP;
+import static org.folio.innreach.domain.dto.folio.inventory.InventoryItemStatus.CHECKED_OUT;
+import static org.folio.innreach.domain.dto.folio.inventory.InventoryItemStatus.DECLARED_LOST;
+import static org.folio.innreach.domain.dto.folio.inventory.InventoryItemStatus.IN_PROCESS;
+import static org.folio.innreach.domain.dto.folio.inventory.InventoryItemStatus.IN_TRANSIT;
+import static org.folio.innreach.domain.dto.folio.inventory.InventoryItemStatus.PAGED;
+import static org.folio.innreach.domain.dto.folio.inventory.InventoryItemStatus.WITHDRAWN;
+import static org.folio.innreach.domain.dto.folio.requeststorage.RequestDTO.RequestStatus;
+import static org.folio.innreach.domain.dto.folio.requeststorage.RequestDTO.RequestStatus.OPEN_AWAITING_DELIVERY;
+import static org.folio.innreach.domain.dto.folio.requeststorage.RequestDTO.RequestStatus.OPEN_AWAITING_PICKUP;
+import static org.folio.innreach.domain.dto.folio.requeststorage.RequestDTO.RequestStatus.OPEN_IN_TRANSIT;
+import static org.folio.innreach.domain.dto.folio.requeststorage.RequestDTO.RequestStatus.OPEN_NOT_YET_FILLED;
+import static org.folio.innreach.domain.dto.folio.requeststorage.RequestDTO.RequestType.HOLD;
+import static org.folio.innreach.domain.dto.folio.requeststorage.RequestDTO.RequestType.PAGE;
+import static org.folio.innreach.domain.dto.folio.requeststorage.RequestDTO.FulfilmentPreference.HOLD_SHELF;
+
 @RequiredArgsConstructor
 @Service
 public class InnReachTransactionServiceImpl implements InnReachTransactionService {
 
   private final InnReachTransactionRepository repository;
+  private final CentralPatronTypeMappingRepository centralPatronTypeMappingRepository;
+  private final CentralServerRepository centralServerRepository;
   private final InnReachTransactionMapper mapper;
   private final CentralServerService centralServerService;
 
+  private final InventoryClient inventoryClient;
+  private final RequestStorageClient requestsClient;
+  private final InventoryStorageClient inventoryStorageClient;
+  private final UsersClient usersClient;
+
+  private static final Set<RequestStatus> openRequestStatuses = Set.of(OPEN_AWAITING_PICKUP, OPEN_AWAITING_DELIVERY,
+    OPEN_IN_TRANSIT, OPEN_NOT_YET_FILLED);
+  private static final Set<InventoryItemStatus> availableItemStatuses = Set.of(
+    AVAILABLE, CHECKED_OUT, PAGED, IN_TRANSIT, DECLARED_LOST, WITHDRAWN, AWAITING_PICKUP, AWAITING_DELIVERY);
```

Comment:
Please correct me, but I understood from the requirements that "available item statuses" are **NOT** one of 

> "Withdrawn", "Missing", "Checked out", "Declared lost", "Aged to lost", or another status configured a "not available" in the settings defined in MODINREACH-81